### PR TITLE
chore: bump deps

### PR DESCRIPTION
**Pull request by kanadgupta**
## 🧰 Changes

- [x] Removed `semantic-release` deps in favor of installing them on the fly in our release workflow
- [x] Ran `npm up` to update a bunch of dependencies

Before (with `semantic-release` deps):

```
Package                          Current   Wanted   Latest  Location                                      Depended by
@commitlint/cli                   17.4.2   17.4.4   17.4.4  node_modules/@commitlint/cli                  rdme
@commitlint/config-conventional   17.4.2   17.4.4   17.4.4  node_modules/@commitlint/config-conventional  rdme
@readme/eslint-config             10.5.0   10.5.1   10.5.1  node_modules/@readme/eslint-config            rdme
@types/parse-link-header           2.0.0    2.0.1    2.0.1  node_modules/@types/parse-link-header         rdme
@types/validator                 13.7.11  13.7.13  13.7.13  node_modules/@types/validator                 rdme
chalk                              4.1.2    4.1.2    5.2.0  node_modules/chalk                            rdme
ci-info                            3.7.1    3.8.0    3.8.0  node_modules/ci-info                          rdme
configstore                        5.0.1    5.0.1    6.0.0  node_modules/configstore                      rdme
eslint                            8.33.0   8.35.0   8.35.0  node_modules/eslint                           rdme
jest                              29.4.1   29.4.3   29.4.3  node_modules/jest                             rdme
jest-extended                      3.2.3    3.2.4    3.2.4  node_modules/jest-extended                    rdme
node-fetch                         2.6.9    2.6.9    3.3.0  node_modules/node-fetch                       rdme
oas                               20.4.0   20.5.1   20.5.1  node_modules/oas                              rdme
open                               8.4.0    8.4.2    8.4.2  node_modules/open                             rdme
ora                                5.4.1    5.4.1    6.1.2  node_modules/ora                              rdme
prettier                           2.8.3    2.8.4    2.8.4  node_modules/prettier                         rdme
semantic-release                  19.0.5   19.0.5   20.1.1  node_modules/semantic-release                 rdme
simple-git                        3.16.0   3.16.1   3.16.1  node_modules/simple-git                       rdme
type-fest                          3.5.4    3.6.1    3.6.1  node_modules/type-fest                        rdme
validator                         13.7.0   13.9.0   13.9.0  node_modules/validator                        rdme
```

Before (with `semantic-release` deps uninstalled):

```
Package                          Current   Wanted   Latest  Location                                      Depended by
@commitlint/cli                   17.4.2   17.4.4   17.4.4  node_modules/@commitlint/cli                  rdme
@commitlint/config-conventional   17.4.2   17.4.4   17.4.4  node_modules/@commitlint/config-conventional  rdme
@readme/eslint-config             10.5.0   10.5.1   10.5.1  node_modules/@readme/eslint-config            rdme
@types/parse-link-header           2.0.0    2.0.1    2.0.1  node_modules/@types/parse-link-header         rdme
@types/validator                 13.7.11  13.7.13  13.7.13  node_modules/@types/validator                 rdme
chalk                              4.1.2    4.1.2    5.2.0  node_modules/chalk                            rdme
ci-info                            3.7.1    3.8.0    3.8.0  node_modules/ci-info                          rdme
configstore                        5.0.1    5.0.1    6.0.0  node_modules/configstore                      rdme
eslint                            8.33.0   8.35.0   8.35.0  node_modules/eslint                           rdme
jest                              29.4.1   29.4.3   29.4.3  node_modules/jest                             rdme
jest-extended                      3.2.3    3.2.4    3.2.4  node_modules/jest-extended                    rdme
node-fetch                         2.6.9    2.6.9    3.3.0  node_modules/node-fetch                       rdme
oas                               20.4.0   20.5.1   20.5.1  node_modules/oas                              rdme
open                               8.4.0    8.4.2    8.4.2  node_modules/open                             rdme
ora                                5.4.1    5.4.1    6.1.2  node_modules/ora                              rdme
prettier                           2.8.3    2.8.4    2.8.4  node_modules/prettier                         rdme
simple-git                        3.16.0   3.16.1   3.16.1  node_modules/simple-git                       rdme
type-fest                          3.5.4    3.6.1    3.6.1  node_modules/type-fest                        rdme
validator                         13.7.0   13.9.0   13.9.0  node_modules/validator                        rdme
```

After:

```
Package      Current  Wanted  Latest  Location                  Depended by
chalk          4.1.2   4.1.2   5.2.0  node_modules/chalk        rdme
configstore    5.0.1   5.0.1   6.0.0  node_modules/configstore  rdme
node-fetch     2.6.9   2.6.9   3.3.0  node_modules/node-fetch   rdme
ora            5.4.1   5.4.1   6.1.2  node_modules/ora          rdme
```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -19,6 +19,8 @@ jobs:
           cache: 'npm'
       - name: Install dependencies and build TS
         run: npm ci && npm run build
+      - name: Install semantic-release and friends
+        run: npm i --no-save semantic-release@19 @semantic-release/changelog@6  @semantic-release/exec@6 @semantic-release/git@10
```
Comment (erunion) [inline .github/workflows/release.yml:23]:
```suggestion
        run: npm i --no-save semantic-release@19 @semantic-release/changelog@6 @semantic-release/exec@6 @semantic-release/git@10
```

**Comment by erunion** [comment]
Mind bumping `oas` again? I just pushed up [v20.5.1](https://github.com/readmeio/oas/releases/tag/20.5.1) with a change to the OAS reducer to make it case-insensitive.

**Comment by kanadgupta** [comment]
@erunion yep thanks!

File: .github/workflows/release.yml (hunk 1 above)
Comment (kanadgupta) [reply .github/workflows/release.yml:23]: oh damn nice catch

**Pull request merged**